torch/distributed/distributed_c10d.py

```diff
@@ -2370,6 +2377,8 @@ def barrier(group=GroupMember.WORLD,
         group (ProcessGroup, optional): The process group to work on. If None,
             the default process group will be used.
         async_op (bool, optional): Whether this op should be an async op
+        devicesId (int or [int], optional): Single or List of device/GPU id
```

Comment:
nit: let's call this `device_ids`. And maybe we should be explicit and always require `List[int]` even when it is a single device?